### PR TITLE
Option to prevent auto shutdown at the end of a run.

### PR DESCRIPTION
**Pull request by hjoliver**
Here's a small change that has been requested by users a number of times: a command line option (and corresponding check-box in the gcylc suite run dialog) that stops the suite daemon from shutting down automatically once all tasks have finished.  The default is to shut down as normal, so this should not result in vast numbers of zombie suite daemons.  This will make it easier to manually retrigger tasks near the end of a suite run, during suite development or debugging.

@matthewrmshin - please review.  If you don't disagree with this, I'll add a new test case.

### CODE REVIEW

File: lib/cylc/gui/app_gcylc.py [hunk 1]
```diff
@@ -1909,7 +1909,11 @@ def startsuite_popup(self, b):
         nodetach_group.pack(hbox)
         vbox.pack_start(hbox)
 
-        optgroups = [nodetach_group, debug_group]
+        noautoshutdown_group = controlled_option_group("No-auto-shutdown", "--no-auto-shutdown")
```
Comment (matthewrmshin) [inline lib/cylc/gui/app_gcylc.py]: This line fails PEP8.

**Comment by matthewrmshin** [comment]
The change looks OK.

**Comment by hjoliver** [comment]
New test added.

**Comment by matthewrmshin** [comment]
@benfitzpatrick please review 2.

**Comment by benfitzpatrick** [comment]
Looks good to me.

**Comment by benfitzpatrick** [comment]
Should we have this as a suite.rc option as well, in a future pull request?

**Comment by hjoliver** [comment]
> Should we have this as a suite.rc option as well, in a future pull request?

I thought the Met Office might be concerned about this getting used as a matter of course?  (=> many suite daemons left running indefinitely when then don't need to be).   Happy to do it if the MO team agrees though.   "Abort on timeout" still overrides this to cause a shutdown.

**Comment by dpmatthews** [comment]
We now configure a site default suite timeout & shutdown so I'm not too worried about this being configured in a suite where appropriate.

**Pull request merged**